immuni_analytics/apis/analytics.py

```diff
@@ -0,0 +1,292 @@
+
+import logging
+from datetime import date
+from http import HTTPStatus
+
+from aioredis.commands import StringCommandsMixin
+from marshmallow import fields
+from mongoengine import StringField
+from sanic import Blueprint
+from sanic.request import Request
+from sanic.response import HTTPResponse
+from sanic_openapi import doc
+
+from immuni_analytics.core import config
+from immuni_analytics.core.managers import managers
+from immuni_analytics.helpers import safety_net
+from immuni_analytics.helpers.redis import get_authorized_tokens_redis_key_current_month
+from immuni_analytics.helpers.safety_net import SafetyNetVerificationError
+from immuni_analytics.models.operational_info import OperationalInfo as OperationalInfoDocument
+from immuni_analytics.models.swagger import (
+    AppleOperationalInfo,
+    AuthorizationBody,
+    GoogleOperationalInfo,
+)
+from immuni_analytics.tasks.authorize_analytics_token import authorize_analytics_token
+from immuni_analytics.tasks.store_operational_info import store_operational_info
+from immuni_common.core.exceptions import SchemaValidationException
+from immuni_common.helpers.sanic import json_response, validate
+from immuni_common.helpers.swagger import doc_exception
+from immuni_common.models.enums import Location, Platform
+from immuni_common.models.marshmallow.fields import (
+    Base64String,
+    IntegerBoolField,
+    IsoDate,
+    Province,
+)
+from immuni_common.models.swagger import HeaderImmuniContentTypeJson
+
+_LOGGER = logging.getLogger(__name__)
+
+bp = Blueprint("analytics", url_prefix="analytics")
+
+
+@bp.route("/apple/operational-info", methods=["POST"], version=1)
+@doc.summary("Upload operational info (caller: Mobile Client.)")
+@doc.description(
+    "Check if the analytics_token is authorized to upload and save the operational info"
+    " in the database."
+)
+@doc.consumes(doc.Boolean(name="Immuni-Dummy-Data", required=True), location="headers")
+@doc.consumes(AppleOperationalInfo, location="body")
+@doc.consumes(HeaderImmuniContentTypeJson(), location="header", required=True)
+@doc.consumes(
+    doc.String(name="Authorization", description="Bearer <ANALYTICS_TOKEN>"),
+    location="header",
+    required=True,
+)
+@doc_exception(SchemaValidationException)
+@doc.response(
+    HTTPStatus.NO_CONTENT.value, None, description="Well-formed request.",
+)
+@validate(
+    location=Location.HEADERS,
+    is_dummy=IntegerBoolField(
+        required=True,
+        data_key="Immuni-Dummy-Data",
+        allow_strings=True,
+        description="Whether the current request is a dummy request. Dummy requests are ignored.",
+    ),
+)
+@validate(
+    location=Location.JSON,
+    province=Province(),
+    exposure_permission=IntegerBoolField(required=True),
+    bluetooth_active=IntegerBoolField(required=True),
+    notification_permission=IntegerBoolField(required=True),
+    exposure_notification=IntegerBoolField(required=True),
+    last_risky_exposure_on=IsoDate(),
```

Comment:
check that the IsoDate validation does not fail in case we are passed a `1970-01-01` date in case of an analytics without exposure

Reply:
The analytics without exposure now sends the current date. Is it stated in some document?